frontend/src/components/new-position-form.js

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+import { moment } from "moment";
+
+const toNames = ({ first_name, last_name }) => `${first_name} ${last_name}`;
+
+const validNumber = val => (val && isNaN(val) ? "Not a valid number" : false);
+const validDate = val =>
+    (val && moment(val, "YYYY-MM-DD", true).isValid()) || val === ""
+        ? false
+        : "Not a valid date";
+// check for format XXX123H1
+const validCourseCode = val => {
+    val = val.toUpperCase();
+    if (
+        val.length === 8 &&
+        isNaN(val.slice(0, 3)) &&
+        !isNaN(val.slice(3, 6)) &&
+        (val[6] === "H" || val[6] === "Y") &&
+        !isNaN(val[7])
+    ) {
+        return false;
+    } else {
+        return "Not a valid course code";
+    }
+};
+const validInstructor = (val, props) => {
+    const instructors = props.instructors.map(toNames);
+    return instructors.indexOf(val) !== -1 ? false : "Not a valid instructor";
+};
+
+export const newPositionFields = [
+    {
+        label: "Course Code",
+        value: "course_code",
+        validate: [validCourseCode],
```

Comment:
Why are the validators an array? Do we need to chain validators?

Reply:
Replaced array with single lambda function